cmd/runu/runu.go

```diff
@@ -84,36 +84,47 @@ func (opts *Runu) PersistentPre(cmd *cobra.Command, _ []string) error {
 	return nil
 }
 
-func (*Runu) Run(cmd *cobra.Command, args []string) error {
+func (*Runu) Run(cmd *cobra.Command, args []string, cfgMgr *config.ConfigManager[config.KraftKit]) error {
 	return cmd.Help()
 }
 
 func main() {
-	cmd := New()
-
 	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
 	defer cancel()
 
 	copts := &cli.CliOptions{}
 
+	// TODO(jake-ciolek): We use the command tree to hydrate config.
+	//                    We use config to hydrate the command tree.
+	//                    This results in a circulard initialization dependency.
+	//                    Right now, we'll instantiate a short-lived, unhydrated
+	//                    command tree to initialize a config manager.
+	//                    Then, we'll use that to obtain the final hydrated command tree.
+	//                    This will do for now, but spend some time thinking about how to make it nicer.
+	cmdInit := New(nil)
+	_, args, err := cmdInit.Find(os.Args[1:])
+	if err != nil {
+		log.G(ctx).Error(err)
+	}
+
+	cfgMgr, err := cli.ConfigManagerFromArgs(args)
```

Comment:
In fact we don't need a config manager in runu. It's purely driven by CLI flags, as per the runc CLI spec. Feel free to get rid of it.

Reply:
Thanks. I've tried to remove that dependency just now, but runu's Create command has a dependency on the config. It's being used by genMachineArgs(). The cmdfactory magic relies on param signatures. For it to work, the params of the commands need to have the specific signature (containing config). We can streamline it further down the line but it will require further work. Could we live with this coupling for now?

Reply:
Argh, right I forgot about that.
Alternatively, could we simply initialize an empty config and pass that wherever KraftKit absolutely requires one? `runu` really isn't meant to inherit KraftKit's config (the fact that it does is a hack right now), so if we strip that dependency to the minimum you might be able to do without a config manager.